kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/rbac/AccessControlService.java

```diff
@@ -0,0 +1,384 @@
+    }
+
+    Set<String> requiredActions = context.getConsumerGroupActions()
+        .stream()
+        .map(a -> a.toString().toLowerCase())
+        .collect(Collectors.toSet());
+
+    return isAccessible("consumer", context.getConsumerGroup(), user, context, requiredActions);
+  }
+
+  public Mono<Boolean> isConsumerGroupAccessible(String groupId) {
+    AccessContext accessContext = AccessContext
+        .builder()
+        .consumerGroup(groupId)
+        .consumerGroupActions(ConsumerGroupAction.VIEW)
+        .build();
+
+    return getCachedUser().map(u -> isConsumerGroupAccessible(accessContext, u));
+  }
+
+  private boolean isSchemaAccessible(AccessContext context, AuthenticatedUser user) {
+    if (context.getSchemaActions().isEmpty()) {
+      log.trace("actions are empty, skipping");
+      return true;
+    }
+
+    if (context.getSchema() == null) {
+      throw new IllegalArgumentException();
+    }
+
+    Set<String> requiredActions = context.getSchemaActions()
+        .stream()
+        .map(a -> a.toString().toLowerCase())
+        .collect(Collectors.toSet());
+
+    return isAccessible("schema", context.getSchema(), user, context, requiredActions);
+  }
+
+  public Mono<Boolean> isSchemaAccessible(String schema) {
+    AccessContext accessContext = AccessContext
+        .builder()
+        .schema(schema)
+        .schemaActions(SchemaAction.VIEW)
+        .build();
+
+    return getCachedUser().map(u -> isSchemaAccessible(accessContext, u));
+  }
+
+  private boolean isConnectAccessible(AccessContext context, AuthenticatedUser user) {
+    if (context.getConnectActions().isEmpty()) {
+      log.trace("actions are empty, skipping");
+      return true;
+    }
+
+    if (context.getConnect() == null) {
+      throw new IllegalArgumentException();
+    }
+
+    Set<String> requiredActions = context.getConnectActions()
+        .stream()
+        .map(a -> a.toString().toLowerCase())
+        .collect(Collectors.toSet());
+
+    return isAccessible("connect", context.getConnect(), user, context, requiredActions);
+  }
+
+  public Mono<Boolean> isConnectAccessible(ConnectDTO dto) {
+    return isConnectAccessible(dto.getName());
+  }
+
+  public Mono<Boolean> isConnectAccessible(String connectName) {
+    AccessContext accessContext = AccessContext
+        .builder()
+        .connect(connectName)
+        .connectActions(ConnectAction.VIEW)
+        .build();
+
+    return getCachedUser().map(u -> isConnectAccessible(accessContext, u));
+  }
+
+
+  public boolean isConnectorAccessible(AccessContext context, AuthenticatedUser user) {
+    return isConnectAccessible(context, user);
+  }
+
+  public Mono<Boolean> isConnectorAccessible(String connectName, String connectorName) {
+    AccessContext accessContext = AccessContext
+        .builder()
+        .connect(connectName)
+        .connectActions(ConnectAction.VIEW)
+        .connector(connectorName)
+        .connectorActions(ConnectorAction.VIEW)
+        .build();
```

Comment:
how it will be placed in yaml file - will connect & connector be on the same level?